libs/header/p2p/exchange_test.go

```diff
@@ -382,6 +382,38 @@ func TestExchange_RequestHeadersFromAnotherPeerWhenTimeout(t *testing.T) {
 	assert.NotEqual(t, newPeerScore, prevScore)
 }
 
+// TestExchange_RequestPartialRange enusres in case of receiving a partial response
+// from server, Exchange will re-request remaining headers from another peer
+func TestExchange_RequestPartialRange(t *testing.T) {
+	hosts := createMocknet(t, 3)
+	exchg, _ := createP2PExAndServer(t, hosts[0], hosts[1])
+
+	// create one more server(with more headers in the store)
+	serverSideEx, err := NewExchangeServer[*test.DummyHeader](
+		hosts[2], headerMock.NewStore[*test.DummyHeader](t, test.NewTestSuite(t), 10),
+		WithNetworkID[ServerParameters](networkID),
+	)
+	require.NoError(t, err)
+	require.NoError(t, serverSideEx.Start(context.TODO()))
+	exchg.peerTracker.peerLk.Lock()
+	prevScoreBefore1 := exchg.peerTracker.trackedPeers[hosts[1].ID()].peerScore
+	prevScoreBefore2 := 50
+	// reducing peerScore of the second server, so our exchange will request host[1] first.
+	exchg.peerTracker.trackedPeers[hosts[2].ID()] = &peerStat{peerID: hosts[2].ID(), peerScore: 50}
+	exchg.peerTracker.peerLk.Unlock()
+	h, err := exchg.GetRangeByHeight(context.Background(), 1, 8)
```

Comment:
different ctxs also makes this look inconsistent

Reply:
oops, yes, will change. Thanks

Reply:
It's still ugly though. One ctx defiend at the top of the test, instead of multiple Background calls